src/runtime/hexagon_remote/sim_remote.cpp

```diff
@@ -40,6 +50,50 @@ int qurt_hvx_unlock() {
     return 0;
 }
 
+int get_hvx_mode() {
+    unsigned int mode;
+    __asm volatile (
+                    " %0 = syscfg\n"
+                    " %0 = and(%0, #128) // v2x\n"
+                    : "=r"(mode)
+                    );
+    __asm volatile (
+                    "isync\n"
+                    );
+
+    return mode;
+}
+
+struct work {
+    void *user_context;
+    uint8_t *closure;
+    halide_task_t f;
+    int min;
+    int max;
+    int result;
+};
+typedef struct work work;
+work work_queue[MAX_WORKER_THREADS];
```

Comment:
I think these should go on the stack in halide_do_par_for. Having them be global seems to be broken for nested parallelism.